File: app/src/main/java/com/sppe/shrimppaste/view/activity/LaunchActivity.java
package com.sppe.shrimppaste.view.activity;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.view.View;

import com.sppe.shrimppaste.R;

import butterknife.ButterKnife;
import butterknife.OnClick;

public class LaunchActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_launch);

        ButterKnife.bind(this);
    }

    @OnClick(R.id.to_main_btn)
    public void onClick(View v) {
        switch (v.getId()){
            case R.id.to_main_btn:
                startActivity(new Intent(LaunchActivity.this,HomeActivity.class));
                Log.e("======","lets go");
                finish();
                break;
        }
    }

}

File: app/src/main/java/com/sppe/shrimppaste/ui/activity/LaunchActivity.kt
package com.sppe.shrimppaste.ui.activity

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity

import com.sppe.shrimppaste.R
import kotlinx.android.synthetic.main.activity_launch.*

/**
 *
 * Created by WangHaoFei on 2017/10/22.
 */
class LaunchActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_launch)

        btn_to_main.setOnClickListener {
            startActivity(Intent(this@LaunchActivity,HomeActivity::class.java))
        }
    }

}
